### PR TITLE
MNT: Run PTH flake test in prep for supporting pathlib (stats)

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Extracted from #16060, ~based off #16917 (only the last commit is exclusive to this branch)~

@larrybradley already approved this bit in https://github.com/astropy/astropy/pull/16060#pullrequestreview-1888630539

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.
Co-authored-by: Mridul Seth <git@mriduls.com>

### CODE REVIEW

**Comment by neutrinoceros** [comment]
Note that labels `Docs`, `testing` and `utils.iers` can be removed (they were due to commit from the branch this was previously based off)

File: astropy/stats/setup_package.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def get_extensions() -> list[Extension, Extension]:
     _stats_ext = Extension(
         name="astropy.stats._stats",
         define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
-        sources=[os.path.join(ROOT, "_stats.pyx")],
+        sources=[str(ROOT / "_stats.pyx")],
```
Comment (nstarman) [inline astropy/stats/setup_package.py:26]: This is OS-dependent.

**Review by nstarman — APPROVED** [review]
LGTM!

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**